### PR TITLE
Move App Services to Swift Packages

### PR DESCRIPTION
**Pull request by skhamis**
Fixes https://github.com/mozilla/application-services/issues/4477

Converts these libraries from Carthage to Swift Packages:
- MozillaAppServices
- SwiftKeychainWrapper (used by MozillaAppServices + Firefox-ios)
- SwiftProtobuf (used by MozillaAppServices + Firefox-ios)
- Telemetry (was easy)

### Note: M1 Macs will not be able to use both Carthage + SPM without Rosetta enabled on Xcode! Carthage-built files will give an error that the binaries were built with x86_64. The solution is for M1s to use rosetta until all packages are SPM
However, given that M1s can't run firefox-ios at all -- it might still be okay to land (when appropriate)

### CODE REVIEW

**Comment by skhamis** [comment]
Looks like trying to fix the merge conflict generated some errors....code should still be sound -- seems like potentially from the swift package itself and will look at it tomorrow

**Comment by skhamis** [comment]
Hi team! How is the review for this going? I'm open to discussing any hesitation with the current hybrid approach and willing to hop on a call if needed to walk through the process!

**Review by electricRGB — APPROVED** [review]
This look good to me. :)

**Comment by skhamis** [comment]
Appreciate the review @electricRGB! Is it possible to also get a Bitrise run just to ensure things are looking good before the merge? Thanks again!

**Comment by nbhasin2** [comment]
@skhamis lets wait till monday while I do one last pass. 

Will check that it builds both locally and on BR it archives successfully and can be pushed to beta. Its important to check that before merging it in.

**Comment by nbhasin2** [comment]
Folks, I marked this as `Do Not Merge` so we don't accidentally merge it until dependency changes are resolved and get an archive. 

I was taking a closer look at the changes in pbxproj and will pair with @skhamis to address them.

File: Client.xcodeproj/project.pbxproj
Comment (nishant2718) [inline Client.xcodeproj/project.pbxproj:65]: I noticed that there's 8 modules from AS out of 10 I see in `Package.swift` [from here](https://github.com/mozilla/rust-components-swift/blob/main/Package.swift). Are the other two not ready to be consumed by iOS?

File: Client.xcodeproj/project.pbxproj
Comment (nishant2718) [inline Client.xcodeproj/project.pbxproj:12165]: Can you explain this? I'm curious about why this works and builds when Carthage is still there for the others.

File: Client.xcodeproj/project.pbxproj
Comment (nishant2718) [inline Client.xcodeproj/project.pbxproj:12479]: Do we eventually have to change this repoURL to `https://github.com/mozilla/rust-components-swift`?

File: Client.xcodeproj/project.pbxproj
Comment (nishant2718) [inline Client.xcodeproj/project.pbxproj:12527]: I'm seeing duplicates in this section, like RustLog and SwiftKeychainWrapper. Is this just some weird XCode thing that's okay to have?

File: Storage/Rust/RustLogins.swift
Comment (nishant2718) [inline Storage/Rust/RustLogins.swift:10]: I think remember seeing somewhere saying an `_` indicates an API that's not production ready. I might be wrong, but are `@_exported import` official?

Comment (nishant2718) [reply Storage/Rust/RustLogins.swift:10]: Oh never mind! I remembered wrong 😅

File: Client.xcodeproj/project.pbxproj
Comment (skhamis) [reply Client.xcodeproj/project.pbxproj:65]: iOS doesn't current use the application-services versions of Push, Tabs, Autofill and CrashTest (not an actual component) so those won't be added until there is a roadmap plan to have iOS switch over to using our versions!

File: Client.xcodeproj/project.pbxproj
Comment (skhamis) [reply Client.xcodeproj/project.pbxproj:12527]: yeah I don't manually do anything in this file it's more or less "automatic" after adjusting settings in code. Since there are quite a few targets + extensions in the project there might look like some overlap

File: Client.xcodeproj/project.xcworkspace/contents.xcworkspacedata
Comment (skhamis) [reply Client.xcodeproj/project.xcworkspace/contents.xcworkspacedata:5]: I blame Xcode!

File: Client.xcodeproj/project.pbxproj
Comment (skhamis) [reply Client.xcodeproj/project.pbxproj:12479]: This is a really good catch! I couldn't find where this was in the project as there is no reference to it so I wonder if it's just an artifact that didn't get cleaned up for some reason...will investigate

File: Client.xcodeproj/project.pbxproj
Comment (skhamis) [inline Client.xcodeproj/project.pbxproj:12519]: @nishant2718 here is the actual reference

**Comment by skhamis** [comment]
Based on a meeting & conversation with the firefox-ios team. This will be better in the hands of someone in their team who will understand the context of how all the deps should be linked. I will be marking my side as done and have a side tracking task to assist when necessary.

**Comment by nbhasin2** [comment]
we merged spm changes to main https://github.com/mozilla-mobile/firefox-ios/pull/9507

**Pull request closed without merging**
